### ISSUE TITLE
Embedder loads in meta-device state under memory pressure — undetected until inference, retries replay broken state

### ISSUE THREAD

**Issue by Cyb3rDudu** [Cyb3rDudu/axiom#266]
## Evidence (production, 2026-09-12, FIN caption wave A first attempt)

7 force-rebuild jobs all failed identically with `RETRY_EXHAUSTED`:

```
embedding batch failed at chunk offset 0 (4 chunks):
Cannot copy out of meta tensor; no data! Please use torch.nn.Module.to_empty()
instead of torch.nn.Module.to() when moving module from meta to a different device.
```

Context: runner had been restarted (workdir cleared), query models (BGE-M3 embedder +
reranker on MPS) loaded during warmup, then the first ingest job loaded the ingest
TextEmbedder — under memory pressure torch initialized weights on the meta device;
the first real inference batch then raised. All retry attempts reused the same broken
in-process model state and failed identically. A second runner restart cleared it;
the same jobs then processed normally.

## Problem

1. The meta-tensor state is **not detected at load time** — the model "loads" fine
   and only explodes at first inference, three times in a row.
2. Failures of this class should reset the model singleton, not just fail the job:
   the retry loop has no chance while the broken model stays resident.

## Fix direction

- After loading a model, run a **tiny inference probe** (one embedded token) and
  treat failure as a load failure (reload once, then loud error) — meta-device
  breakage becomes a load-time error instead of a job-time one.
- On an embedding-batch error, **drop the embedder singleton** so the next attempt
  reloads from scratch instead of replaying the broken state.

## Definition of Done

- [ ] Unit/IT: a meta-device-initialized model (or a stub raising at first
      inference) is caught by the load probe, reloaded, and the job succeeds or
      fails loudly at load time — not at batch time
- [ ] On batch failure the singleton is dropped (test pins the reload on retry)
- [ ] Evidence: re-run of a caption wave after a cold runner restart processes
      without the 3× identical failure

## Non-goals

- Solving MPS memory pressure itself (that is capacity planning, not product)

**Comment by Cyb3rDudu** [comment]
Landed on main (c9626bc, strand A): load probe + warm-singleton drop. Live evidence: the 2026-09-12 incident class stayed silent through all subsequent waves including the 146-lecture night and tonight's E2E under full memory (embedder+reranker+GLiNER+mREBEL resident).
